### ISSUE TITLE
[Rule Request] `react-hooks/prefer-use-state-lazy-initialization`

### ISSUE THREAD

**Issue by SukkaW** [Rel1cx/eslint-react#214]
### Describe the problem

Disallow function calls in `useState` that aren't wrapped in an initializer function:

```ts
// Bad
const [value, setValue] = useState(generateTodos());
// Good
const [value, setValue] = useState(() => generateTodos());
```

### Describe the solution you'd like

Port https://github.com/jsx-eslint/eslint-plugin-react/pull/3579
